### DISCUSSION TITLE
сложность игры и свитки

### DISCUSSION THREAD

**Discussion by Kor-Angar** [dosinabox/g2nr_unofficial_update#350] [Show and tell]
играю в Готику 2 классик и местами она конечно Готики 2 НВ. особенно на свитках это чувствуется. в связи с этим вопрос, а нельзя ли в неоф. обновление опционально сделать что бы свитки требовали столько же маны сколько требуют и руны? технически это возможно?

Было бы интересно поиграть с такими настройками и в плане баланса (усложнения игры) и в плане "конона".

**Comment by dosinabox** [comment]
Это не получится сделать настройкой. Только отдельные скрипты.
И как быть со свитками превращения? У них ведь нет рун.

**Reply by Kor-Angar** [reply]
> Это не получится сделать настройкой. Только отдельные скрипты.

жаль

**Comment by Kor-Angar** [comment]
> И как быть со свитками превращения? У них ведь нет рун.

посмотреть на стоимость в Готике 2 Классик

***

- Овечка - 5
- Крыса - 5
- Падальщик - 5
- Полевой жук - 5
- Луркер - 5 (странно что не 10)


***

- Волк - 10 (странно что не 5)
- Снеппер - 10
- Варг - 10
- Варан(Ящерица) - 10

***

- Дракон Снеппер - 15
- Огненная Ящерица - 15




![image](https://user-images.githubusercontent.com/49514764/87795626-03360700-c851-11ea-89c6-91a26e9ef353.png)

**Reply by Kor-Angar** [reply]
> посмотреть на стоимость в Готике 2 Классик

Хотя расоход маны в Готике 2 классик ниже в 2-3 раза относительно Готики 2 НВ. поэтому наверное необходимое кол-во маны на свиток превращения можно увеличить. Например

  | Готика 2 классик | Готика 2 НВ | Готика 2 НВ | Готика 2 НВ
-- | -- | -- | -- | --
  | _ | Варинат 1 | Варинат 2 | Варинат 3
простые | 5 маны | 10 | 10 | 15
средния | 10 маны | 20 | 25 | 30
сложные | 15 маны | 30 | 45 | 50

**Reply by Kor-Angar** [reply]
> @KlingenGrossmeister  : Третий вариант самый нормальный - заставит во второй главе обзавестись в Яркендаре бижутерией, либо бегать и собирать грибы, либо попытаться собрать хотя бы 10 старых каменных таблиц. Плюс нет возможности зачистить полХориниса варгом, огненным вараном и драконьим снеппером.

Третий вариант мягко говоря драконовский по расценкам :-) и герой наверное в первой главе 50 маны без прокачки не наберёт. Можно начать например с первого варианта, поиграть, а потом если что еще усложнить.

**Reply by Kor-Angar** [reply]
подсказка: что бы определить "правильную" стоимости маны для свитков превращения можно сапоставить расходы маны в  Готике 2 Классик и НВ у других заклинаний. 

  | Готки 2 Классик | Готика 2 НВ
-- | -- | --
превращение простое | 5 |  
лёгкое лечение | 5 | 10
превращение среднее | 10 |  
сон | 10 | 30
превращение сложное | 15 |  
страх | 15 | 50

**Comment by Kor-Angar** [comment]
> Это не получится сделать настройкой. 

а если в файлы Spell_ххх.d добавить проверку на опцию xxxMode  ?


если опция выключена цена идёт свитка, а если опция включена, то цена идёт единая и для свитка и для руны). На примере огненного дождя.

```
func int Spell_Logic_Firerain(var int manaInvested)
{
	if(Npc_GetActiveSpellIsScroll(self) && (self.attribute[ATR_MANA] >= SPL_Cost_Scroll) && (xxxMode == False))
	{
		return SPL_SENDCAST;
	}
	else if(self.attribute[ATR_MANA] >= SPL_Cost_Firerain)
	{
		return SPL_SENDCAST;
	}
	else
	{
		return SPL_SENDSTOP;
	};
};

func void Spell_Cast_Firerain()
{
	if((Npc_GetActiveSpellIsScroll(self))&& (xxxMode == False))
	{
		self.attribute[ATR_MANA] -= SPL_Cost_Scroll;
	}
	else
	{
		self.attribute[ATR_MANA] -= SPL_Cost_Firerain;
	};
	self.aivar[AIV_SelectSpell] += 1;
};

```
попробовал у себя, вроде скрипт сработал.

**Comment by dosinabox** [comment]
Это да, но как быть с описаниями свитков? У них всегда будет написано 5 маны, потому что эти значения - константы и записываются прямо в свойства созданных предметов.

**Comment by Kor-Angar** [comment]
расходы маны на "забыть"(SPL_Cost_Charm) можно сделать такой же как и у сна 30.

сейчас SPL_Cost_Charm = 50 :-) при том что руны то нет :-))

**Reply by Kor-Angar** [reply]
момент актуален


  | Готки 2 Классик | Готика 2 НВ
-- | -- | --
сон | 10 | 30
забыть | 10 | 50

**Reply by Kor-Angar** [reply]
```D
// расход маны у заклинания "забыть" устанавливается такой же как и у заклиния "сон".
const int SPL_Cost_Charm = SPL_Cost_Sleep;
```

**Comment by Kor-Angar** [comment]
гм... а в g2mdk-2.6 расход маны на свитки превращения стоит 10... а у Акеллы точно 5?

```
// *********************
// Alle Transform Spells
// *********************

const int SPL_Cost_TrfSheep			= 10;
const int SPL_Cost_TrfScavenger		= 10;
const int SPL_Cost_TrfGiantRat		= 10;
const int SPL_Cost_TrfGiantBug		= 10;
const int SPL_Cost_TrfWolf			= 10;

const int SPL_Cost_TrfLurker	        	= 10;
const int SPL_Cost_TrfSnapper		= 10;
const int SPL_Cost_TrfWaran			= 10;
const int SPL_Cost_TrfWarg			= 10;

const int SPL_Cost_TrfFireWaran		= 10;
const int SPL_Cost_TrfShadowbeast	= 10;
const int SPL_Cost_TrfDragonSnapper	= 10;
```

**Reply by Kor-Angar** [reply]
тут явно пираньи сами себя запутали. если у всех 5 то почему у этих 10? :-)

**Comment by UnresolvedExternal** [comment]
> Это да, но как быть с описаниями свитков? У них всегда будет написано 5 маны, потому что эти значения - константы и записываются прямо в свойства созданных предметов.

Что мешает записать туда результат вызова функции? Правда уже созданные свитки обновятся только после перезагрузки сейва.

**Comment by dosinabox** [comment]
> На примере огненного дождя.

У заклинаний превращения нет Spell_Cast_*.

> Что мешает записать туда результат вызова функции?

Ничего, но если всё это только для теста, то проще просто изменить существующие константы.

**Comment by dosinabox** [comment]
> гм... а в g2mdk-2.6 расход маны на свитки превращения стоит 10... а у Акеллы точно 5?

5 - стандартное требование маны для свитков (SPL_Cost_Scroll).
10 - требования для заклинаний превращения (SPL_Cost_Trf*). В игре они не используются, потому что рун превращения нет.

**Comment by Kor-Angar** [comment]
> 10 - требования для заклинаний превращения (SPL_Cost_Trf*). В игре они не используются, потому что рун превращения нет.

давай я просто опубликую код из g2mdk-2.6, что бы мы не множили сущности:-)

```
// *********************
// Alle Transform Spells
// *********************

const int SPL_Cost_TrfSheep			= 10;
const int SPL_Cost_TrfScavenger		= 10;
const int SPL_Cost_TrfGiantRat		= 10;
const int SPL_Cost_TrfGiantBug		= 10;
const int SPL_Cost_TrfWolf			= 10;
const int SPL_Cost_TrfWaran			= 10;
const int SPL_Cost_TrfSnapper		= 10;
const int SPL_Cost_TrfWarg			= 10;
const int SPL_Cost_TrfFireWaran		= 10;
const int SPL_Cost_TrfLurker		= 10;
const int SPL_Cost_TrfShadowbeast	= 10;
const int SPL_Cost_TrfDragonSnapper	= 10;

// ------ Instanz fьr alle Transform-Sprьche ------
INSTANCE Spell_Transform (C_Spell_Proto)
{
	time_per_mana			= 0;
	spelltype 				= SPELL_NEUTRAL;
	targetCollectAlgo		= TARGET_COLLECT_NONE;
	canTurnDuringInvest		= 0;
};

// ------ Schaf ------
func int Spell_Logic_TrfSheep (var int manaInvested)
{
	if ((Npc_GetActiveSpellIsScroll(self) && (self.attribute[ATR_MANA] >= SPL_Cost_Scroll)))
	|| (self.attribute[ATR_MANA] >= SPL_Cost_TrfSheep)
	{
		self.attribute[ATR_MANA] = self.attribute[ATR_MANA] - SPL_Cost_TrfSheep;
		
		Npc_SetActiveSpellInfo(self, Sheep);
		return SPL_SENDCAST;
	}
	else
	{
		return SPL_SENDSTOP;
	};
};
```
мы видим что в оригинальных скриптах заклинания превращения фактически отнимают 10 маны, т.к. в них не используется "SPL_Cost_Scroll". в этих скриптах такая же история
https://github.com/dosinabox/g2nr_unofficial_update/blob/d6a0bc1ecebbb3aad43ba9278a9a11820a3ca03f/PrjGOTHIC/AI/Magic/Spells/Spell_Transform.d#L36
только значение SPL_Cost_Trf** изменено с 10 до 5.

**Comment by dosinabox** [comment]
Какое-то старое изменение. Одно из первых.

> Исправлены несоответствия в стоимости использования заклинаний превращения (взято из DNdR Script Patch).

**Comment by dosinabox** [comment]
> Привет. Так получится добавить опционально в тестовой версии? :)


Привет, завтра сделаю. Определись с вариантом.

**Comment by Kor-Angar** [comment]
> Определись с вариантом.

если про свитки превращения то "вариант 2"
```
// *********************
// Alle Transform Spells
// *********************

const int SPL_Cost_TrfSheep			= 10;
const int SPL_Cost_TrfScavenger		= 10;
const int SPL_Cost_TrfGiantRat		= 10;
const int SPL_Cost_TrfGiantBug		= 10;
const int SPL_Cost_TrfWolf			= 10;

const int SPL_Cost_TrfLurker	        	= 25;
const int SPL_Cost_TrfSnapper		= 25;
const int SPL_Cost_TrfWaran			= 25;
const int SPL_Cost_TrfWarg			= 25;

const int SPL_Cost_TrfFireWaran		= 45;
const int SPL_Cost_TrfShadowbeast	= 45;
const int SPL_Cost_TrfDragonSnapper	= 45;
```

**Comment by dosinabox** [comment]
Вот:
https://dropmefiles.com/vyWpH

Если нужно, меняй в свежих скриптах эти константы, свитки теперь используют именно их:
```
const int SPL_Cost_TrfSheep = 5;
const int SPL_Cost_TrfScavenger = 5;
const int SPL_Cost_TrfGiantRat = 5;
const int SPL_Cost_TrfGiantBug = 5;
const int SPL_Cost_TrfWolf = 5;
const int SPL_Cost_TrfWaran = 5;
const int SPL_Cost_TrfSnapper = 5;
const int SPL_Cost_TrfWarg = 5;
const int SPL_Cost_TrfFireWaran = 5;
const int SPL_Cost_TrfLurker = 5;
const int SPL_Cost_TrfShadowbeast = 5;
const int SPL_Cost_TrfDragonSnapper = 5;
```

**Comment by dosinabox** [comment]
уже включено - константы изменены

**Comment by Kor-Angar** [comment]
наверное мы как обычно про разное говорим. я предложил сделать расход маны на свитки  в игре, как это было  в Готике 1 и Готики 2. что бы усложнить таким образом игру. в таком случае герой первого уровня не сможет например убить всех паладинов в порту или всех бандитов в лагере Декстера через свиток огненный дождь. в последствие когда герой прокачает ману или найдёт бонусы к мане и соберет/купки шмотки на ману он сможет использовать сложные свитки типа огненный дождь, вызов демона. но не в начале игры.

Для этого мною было предложено ввести переменную, что бы это можно протестировать, сравнить, понять целесообразность, определить узкие моменты и прочее.

у тебя же почему то всё это свелось к свиткам превращения :-(

**Comment by dosinabox** [comment]
Сказываются события последних дней в моей стране и постоянные нервные недосыпы. Пока не до экспериментов. Поэтому релиз того что есть будет через неделю. По этой же причине в нем не будет ребаланса оружия в продаже.

**Reply by Kor-Angar** [reply]
теперь надо крепко нам спать :-/

**Reply by dosinabox** [reply]
Да уж, с тех пор легче и не стало

**Reply by Kor-Angar** [reply]
наверное легче уже и не будет никогда. точнее мы этого не застанем.

**Comment by Kor-Angar** [comment]
сделал у себя константы в IT_Scrolls.d
```
const int SPL_SC_Cost_PalLight = 5;
const int SPL_SC_Cost_PalLightHeal = 5;
const int SPL_SC_Cost_PalHolyBolt = 5;
```
сделал func void SPL_SC_Cost которая меняла бы стоимость
```
func void SPL_SC_Cost()
{	
	if (ScrollMode == True)
	{
		SPL_SC_Cost_PalLight =	SPL_Cost_PalLight ;
		SPL_SC_Cost_PalLightHeal =	SPL_Cost_PalLightHeal ;
		SPL_SC_Cost_PalHolyBolt =	SPL_Cost_PalHolyBolt ;
	}
	else
	{
		SPL_SC_Cost_PalLight =	SPL_Cost_Scroll ;
		SPL_SC_Cost_PalLightHeal =	SPL_Cost_Scroll ;
		SPL_SC_Cost_PalHolyBolt =	SPL_Cost_Scroll ;
	};
};
```
подскажите пожалуйста куда лучше воткнуть  вызов
SPL_SC_Cost();

а то у меня ума нет и я её в SleepABit.d засунул, что наверное не очень правильно :-) хотя и работает :-)

**Comment by dosinabox** [comment]
> сделал у себя константы в IT_Scrolls.d

Суть констант в том, что они определяются только один раз и дальше только используются без переопределений. Поэтому здесь нужны не const, а var. И изначально без определений, они будут позже:
```
var int SPL_SC_Cost_PalLight;
var int SPL_SC_Cost_PalLightHeal;
var int SPL_SC_Cost_PalHolyBolt;
```

> подскажите пожалуйста куда лучше воткнуть вызов

У меня есть функция B_CheckDynamicText(), которая вызывается при каждой инициализации каждого мира (при переходе из другого или при загрузке сохранения). Добавляй свой код туда.

**Reply by Kor-Angar** [reply]
в итоге вызов SPL_SC_Cost() добавил в Startup.d под B_CheckDynamicText(),

> Поэтому здесь нужны не const, а var

спасибо!

**Reply by Kor-Angar** [reply]
> Заметь, что добавлять нужно четыре раза: в init каждого мира.

там же есть общий INIT_Global который загружается перед загрузкой init_мира.

в принципе у тебя в последней версии B_CheckDynamicText находится как раз INIT_Global

```D
func void INIT_Global()
{
	B_Check_Version();
	B_SetHeroSkin();
	B_CheckDynamicText();
	B_InitMonsterAttitudes();
	B_InitNpcGlobals();
};

```

**Reply by dosinabox** [reply]
Да, теперь так. С тех пор удалось хорошо оптимизировать структуру Startup.d.

**Reply by Kor-Angar** [reply]
> Да, теперь так. С тех пор удалось хорошо оптимизировать структуру Startup.d.

и это хорошо. INIT_Global и Startup_Global становится намного понятнее с комментами из MDK.  если бы ты иногда писала комменты было бы ещё понятнее :-)

**Comment by Kor-Angar** [comment]
угу, я через поиск так сказать :-)  вот только при старте игры не отображается на свитках сколько надо маны. ![image](https://user-images.githubusercontent.com/49514764/93943588-81c09080-fd3b-11ea-8e8f-7202681d5754.png)
приходится сохранить игру, а потом загрузить. но я думаю в процессе осознаю что не так :-)

**Comment by dosinabox** [comment]
> кстати а что это заклинание лечения нельзя произнести если жизни полные? :-)

а зачем? в Г1 тоже нельзя.

**Reply by Kor-Angar** [reply]
***
>> кстати а что это заклинание лечения нельзя произнести если жизни полные? :-)

> а зачем? в Г1 тоже нельзя.

в Готике 1 у исцеления другая механика ;-)

**Comment by Kor-Angar** [comment]
а если я в скриптах свитков поменяю штатную функцию Npc_GetActiveSpellIsScroll на   B_NpcGetActiveSpellIsScroll, то так будет корректно её написать?

**upd**
итоговая версия
```D
var int SpellIsScrollMode;

func int C_NpcGetActiveSpellIsScroll (var C_NPC slf)
{
	if SpellIsScrollMode == 1
	&& Npc_GetActiveSpellIsScroll(slf)
	{
			return TRUE;
	};
	return FALSE;
};
```

**Reply by dosinabox** [reply]
А в чем задумка?

**Reply by Kor-Angar** [reply]
задумка уйти от проверки которая написана выше ([on 2 Aug 2020](https://github.com/dosinabox/g2nr_unofficial_update/discussions/350#discussioncomment-878292)) и добиться того же эффекта с минимиальным изменением кода в файлах самих свитках Spell_****

то есть я в файлах Spell_**** хочу использовать B_NpcGetActiveSpellIsScroll и таким образом опционально включать и выключать  проверки на Npc_GetActiveSpellIsScroll

**Reply by Kor-Angar** [reply]
так вроде работает
```D
func int B_NpcGetActiveSpellIsScroll (var C_NPC slf)
//func int B_NpcGetActiveSpellIsScroll (var int slf)
//func void  B_NpcGetActiveSpellIsScroll (var C_NPC slf)
{
	if SpellIsScrollMode == 1
	{
		Npc_GetActiveSpellIsScroll(slf);
	};
	return FALSE;
};
```

**Reply by dosinabox** [reply]
`func int` вернет какое-то значение, которое можно использовать для проверок и дальнейших расчетов. В твоем случае всегда будет возвращаться 0 (FALSE). Если никакое значение не нужно и проверки не подразумеваются, то используй `func void`, который не требует return:
```cpp
func void B_NpcGetActiveSpellIsScroll(var C_NPC slf)
{
	if(SpellIsScrollMode == 1)
	{
		Npc_GetActiveSpellIsScroll(slf);
	};
};
```

**Reply by Kor-Angar** [reply]
Npc_GetActiveSpellIsScroll это func int которая используется как раз для проверок в Spell_****  и я хотел использовать вместо неё B_NpcGetActiveSpellIsScroll. 

> В твоем случае всегда будет возвращаться 0 (FALSE). 

а если так?
```D
func int B_NpcGetActiveSpellIsScroll (var C_NPC slf)
{
	if SpellIsScrollMode == 1
       && Npc_GetActiveSpellIsScroll(slf)
	{
			return TRUE;
	};
	return FALSE;
};
```

**Reply by dosinabox** [reply]
Так хорошо. Только обычно названия таких функции начинаются с **C_**. **B_** это для `func void`.

**Reply by dosinabox** [reply]
```cpp
// проверяет, является ли активное заклинание у НПС npc свитком, если да – возвращает 1, иначе – 0.
func int Npc_GetActiveSpellIsScroll(var C_NPC npc); 
```

**Reply by Kor-Angar** [reply]
> Только обычно названия таких функции начинаются с C_. B_ это для func void.

раскрылась тайна префиксов :-)


```D
var int SpellIsScrollMode;

func int C_NpcGetActiveSpellIsScroll (var C_NPC slf)
{
	if SpellIsScrollMode == 1
	&& Npc_GetActiveSpellIsScroll(slf)
	{
			return TRUE;
	};
	return FALSE;
};
```

**Comment by Kor-Angar** [comment]
@UnresolvedExternal  а внешнию функцию **Npc_GetActiveSpellIsScroll**  через юнион как то нельзя отредактировать?

**Reply by Kor-Angar** [reply]
написал [Slavemaster](https://worldofplayers.ru/members/107921/) в ЛС на WoP.ru :-)

**Reply by Kor-Angar** [reply]
[Slavemaster](https://worldofplayers.ru/members/107921/) : Подменяет адрес на С++ функцию:

```D
namespace NAMESPACE
{
    int __cdecl Npc_GetActiveSpellIsScroll()
    {
        oCNpc* npc;
        ZARGS(npc);
        ZRETURN(false);
        return 0;
    }

    Sub changeFunc(ZSUB(GameEvent::Init), []
        {
            if (Symbol symbol{ parser, "Npc_GetActiveSpellIsScroll" })
                if (symbol.GetType() == Symbol::Type::ExternalFunc)
                    symbol.GetSymbol()->single_intdata = reinterpret_cast<int>(&Npc_GetActiveSpellIsScroll);
        });
}
```

[zNoActiveScroll.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/8513343/zNoActiveScroll.zip)

**Comment by Kor-Angar** [comment]
переписал всё с чистого листа. сделал в виде инъекции собрал в отдельный том. попробовал на последней неоф. обновление, вроде работает :-) если бы человек жил дольше, то человечество бы уже космос осваивало и/или Землю расхерачило :-)

**Reply by Kor-Angar** [reply]
а так пока поймёшь что и как , чему то научишься, обрётешь необходимый опыт без которого сами знание не имеют ценности, так уже и умирать пора :-) 
***
у нас тут на днях двигатели  советские сняли с ил-96 видимо увезли изучать или просто освежить, что бы выдать за вновь изготовленные.. крепка была советская власть :=)

**Comment by Kor-Angar** [comment]
почему то русская версия неоф. обновление запущенная через мастерскую ругается на свитки лечения паладина
![image](https://user-images.githubusercontent.com/49514764/163711528-138763b2-e968-49b9-a092-764b7640d3a2.png)
а английская не ругается :-)

**Reply by dosinabox** [reply]
В русской версии другие названия свитков:
ItSc_PalHeal_01
ItSc_PalHeal_02
ItSc_PalHeal_03

**Reply by Kor-Angar** [reply]
угу.. я смотрел ветку "мастер" на гитхабе, а сейчас посмотрел "ворк" , а там и правда иначе. игра в перебивание названий инстанций всегда играет злую шутку. и всегда потом :-)

**Reply by Kor-Angar** [reply]
только теперь не понятно что делать со свитками лечения паладина :-)

**Reply by Kor-Angar** [reply]
дайте мне медальку "огородных дел мастер"
```D
/*******************************************************************************************/
Prototype ItSc_PalHeal (C_Item)
{
	name 				=	NAME_Spruchrolle;

	mainflag 			=	ITEM_KAT_RUNE;
	flags 				=	ITEM_MULTI;

	material			=	MAT_LEATHER;

	//spell				= 	SPL_PalHeal;
	cond_atr[2]   		= 	ATR_MANA_MAX;
	//cond_value[2]  		= 	SPL_Cost_Scroll;

	wear				= 	WEAR_EFFECT;
	effect				=	"SPELLFX_WEAKGLIMMER";

	//description			=	NAME_SPL_PalHeal;
	
	TEXT	[0]			=	Name_PaladinScroll	;
};
func void B_ItScTEXTCOUNT()
{
	item.TEXT	[1]			=	NAME_Mana_needed;
	item.COUNT	[1]			=	item.cond_value[2];
	
	item.TEXT	[2]			=	NAME_HealingPerCast;		
	// COUNT	[2]			=	;
	
	item.TEXT	[5]			=	NAME_Value;					
	item.COUNT	[5]			=	item.value;	
};
/*******************************************************************************************/
INSTANCE ItSc_PalLightHeal (ItSc_PalHeal)
{
	value 				=	Value_Sc_PalLightHeal;
	visual				=	"ItSc_PalLightHeal.3DS";
	spell				= 	SPL_PalLightHeal;
	cond_value[2]  		= 	SPL_Cost_PalLightHeal;
	description			=	NAME_SPL_PalLightHeal;
	
	COUNT	[2]			=	SPL_Heal_PalLightHeal;
	B_ItScTEXTCOUNT();
};
/*******************************************************************************************/
INSTANCE ItSc_PalMediumHeal (ItSc_PalHeal)
{
	value 				=	Value_Sc_PalMediumHeal;
	visual				=	"ItSc_PalMediumHeal.3DS";
	spell				= 	SPL_PalMediumHeal;
	cond_value[2]  		= 	SPL_Cost_PalMediumHeal;
	description			=	NAME_SPL_PalMediumHeal;
	
	COUNT	[2]			=	SPL_Heal_PalMediumHeal;
	B_ItScTEXTCOUNT();
};
/*******************************************************************************************/
INSTANCE ItSc_PalFullHeal (ItSc_PalHeal)
{
	value 				=	Value_Sc_PalFullHeal;
	visual				=	"ItSc_PalFullHeal.3DS";
	spell				= 	SPL_PalFullHeal;
	cond_value[2]  		= 	SPL_Cost_PalFullHeal;
	description			=	NAME_SPL_PalFullHeal;		
	
	COUNT	[2]			=	SPL_Heal_PalFullHeal;				
	B_ItScTEXTCOUNT();
};
/*******************************************************************************************/
//		 у D36 новые инстанции, поэтому в неоф. обновление будет ошибка
/*******************************************************************************************/
/*
instance  ItSc_PalLightHeal (C_Item)
{
        ItSc_PalLightHeal_old();
        cond_value[2]                        =        SPL_Cost_PalLightHeal;
        COUNT        [1]                        =        cond_value[2];
};
instance  ItSc_PalMediumHeal (C_Item)
{
        ItSc_PalMediumHeal_old();
        cond_value[2]                        =        SPL_Cost_PalMediumHeal;
        COUNT        [1]                        =        cond_value[2];
};

instance  ItSc_PalFullHeal (C_Item)
{
        ItSc_PalFullHeal_old();
        cond_value[2]                        =        SPL_Cost_PalFullHeal;
        COUNT        [1]                        =        cond_value[2];
};
*/
/*******************************************************************************************/
//		 новые инстанции для неоф. обновления
/*******************************************************************************************/
instance  ItSc_PalHeal_01 (C_Item)
{
	ItSc_PalLightHeal();
};
instance  ItSc_PalHeal_02 (C_Item)
{
	ItSc_PalMediumHeal();
};
instance  ItSc_PalHeal_03 (C_Item)
{
	ItSc_PalFullHeal();
};
```

**Comment by Kor-Angar** [comment]
соберу свои отзывы тут :-)

**Reply by Kor-Angar** [reply]
Author [Kor-Angar](https://github.com/Kor-Angar)
[on 31 Dec 2020](https://github.com/dosinabox/g2nr_unofficial_update/discussions/350#discussioncomment-878321)

в 4 главе у ледяного дракона у силовика ОнД маны не хватило на то что бы задействовать огненный дождь. видимо оттого что ОНД не получал бонус к мане за молитву + у меня не было амулету на ману (но был посох) + я не покупал эликсиры к мане.

в целом вот это действительно способно усложнить игру :-) не виртуально, а реально. при этом баланс то Пираней. и идея Пираней. и концепция тоже Пираней. и главное что теперь понятно, что никакой там левый НПС не может гипотетически просто взять свиток и что то там сделать. так как у него маны просто не хватит.. даже на банальный ледяной блок :-)

**Reply by Kor-Angar** [reply]
[Kor-Angar](https://github.com/Kor-Angar)
[on 14 Sep 2020](https://github.com/dosinabox/g2nr_unofficial_update/discussions/350#discussioncomment-878313)

поиграл я сегодня с правками на свитки. пока впечатления исключительно положительные. в начале игры герой ограничен в свитках и не может воспользоваться даже свитком заморозки. что бы его задействовать без прокачки маны за ЛП надо одеть два кольца и найти первые 50 грибов :-)
в тоже время если герой планирует идти в маги и в начале игры готов вкладывать ЛП в ману получает наоборот на старте бонус: можно свитки использовать :-)
ну и конечно же ни о каких огненных дождях в первой главе и речи не может идти :-)

**Reply by Kor-Angar** [reply]
ссылка на мой тематический срач в вк https://vk.com/feed?w=wall-5819093_930836_r931223

**Comment by Kor-Angar** [comment]
ТЕСТ:

@dosinabox  проверь плиз будет ли корректно работать на неоф. обновление. у меня вроде работает.

[zNoActiveScroll.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/8515731/zNoActiveScroll.zip)

**Reply by dosinabox** [reply]
Что нужно делать и какие результаты ожидаются?

**Reply by Kor-Angar** [reply]
**Что нужно делать**
1) кастануть какой нибудь свиток, например вызов демона или огненный дождь.
2) посмотреть стоимость маны в описание свитков. 
3) соответствует ли расход маны у свитков описанию.

**Reply by Kor-Angar** [reply]
ожидаемые результат:

- свитки расходуют столько же маны что и руны
- в описание у свитков корректно отображается необходимое кол-во маны

**Reply by Kor-Angar** [reply]
upd

[zNoActiveScroll.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/8515731/zNoActiveScroll.zip)

**Reply by dosinabox** [reply]
на днях протестирую из мастерской

**Reply by dosinabox** [reply]
работает

**Reply by Kor-Angar** [reply]
отлично

**Reply by dosinabox** [reply]
Не запускается в классическом моде:
![image](https://user-images.githubusercontent.com/39790055/164946193-875249b7-f58e-4303-ab5a-ef3152c3ab4a.png)

**Reply by Kor-Angar** [reply]
н там и не нужен :-)
